pycrostates/segmentation/entropy.py

```diff
@@ -0,0 +1,584 @@
+    if isinstance(segmentation, _BaseSegmentation):
+        labels = segmentation._labels
+        # reshape if epochs (returns a view)
+        labels = labels.reshape(-1)
+    if isinstance(segmentation, np.ndarray):
+        if not segmentation.ndim == 1:
+            raise ValueError(f"{item_name} must be a 1D array.")
+        if not np.issubdtype(segmentation.dtype, np.integer):
+            raise ValueError(f"{item_name} must be an array of integers.")
+        labels = segmentation
+    return labels
+
+
+@fill_doc
+def _joint_entropy_history(
+    labels: NDArray[int],
+    k: int,
+    state_to_ignore: Optional[int] = -1,
+    log_base: Union[float, str] = 2,
+):
+    r"""Compute the joint Shannon of k-histories x[t:t+k].
+
+    Compute the joint Shannon entropy of the k-histories x[t:t+k].
+
+    Parameters
+    ----------
+    %(labels_info)s
+    %(state_to_ignore)s
+    %(log_base)s
+
+    Returns
+    -------
+    h : float
+        The Shannon entropy of the sequence labels[t:t+k].
+    """
+    _check_type(labels, (np.ndarray,), "labels")
+    _check_type(k, ("int",), "k")
+    if state_to_ignore is not None:
+        state_to_ignore = _ensure_int(state_to_ignore, "state_to_ignore")
+    log_base = _check_log_base(log_base)
+
+    # Construct the k-history sequences while ignoring the state
+    histories = []
+    for i in range(len(labels) - k + 1):
+        history = tuple(labels[i : i + k])
+        if state_to_ignore not in history:
+            histories.append(history)
+
+    n_clusters = np.max(labels) + 1
+    # Compute the joint probability distribution
+    joint_dist = np.zeros(tuple(k * [n_clusters]))
+    for i in range(len(labels) - k + 1):  # TODO: check +1 (not in original code)
+        history = tuple(labels[i : i + k])
+        if state_to_ignore not in history:
+            joint_dist[history] += 1.0
+    # Compute the joint entropy
+    _joint_entropy = scipy.stats.entropy(joint_dist.flatten(), base=log_base)
+    return _joint_entropy
+
+
+@fill_doc
+def _entropy(
+    labels: NDArray[int],
+    state_to_ignore: Optional[int] = -1,
+    log_base: Union[float, str] = 2,
+):
+    r"""Compute the Shannon entropy of the a symbolic sequence.
+
+    Parameters
+    ----------
+    labels: array
+        Symbolic sequence.
+    %(state_to_ignore)s
+    %(ignore_self)s
+    %(log_base)s
+
+    Returns
+    -------
+    h : float
+        The Shannon entropy of the sequence.
+    """
+    _check_type(labels, (np.ndarray,), "labels")
+    if state_to_ignore is not None:
+        _ensure_int(state_to_ignore, "state_to_ignore")
+    log_base = _check_log_base(log_base)
+
+    h = _joint_entropy_history(
+        labels, k=1, state_to_ignore=state_to_ignore, log_base=log_base
+    )
+    return h
+
+
+@fill_doc
+def entropy(
+    segmentation: [Segmentation, NDArray[int]],
+    state_to_ignore: Optional[int] = -1,
+    ignore_self: bool = False,
+    log_base: Union[float, str] = 2,
```

Comment:
I fixed the type-hints/docstring for those 3 arguments. Previously, it was:

```
state_to_ignore: Optional[Union[int, None]]
ignore_self: Optional[bool]
log_base: Optional[Union[float, str]]
```

The first one, I think it's a tpyo.
The 2 others, if None is provided one of the checker would raise.